ros2_control_test_nodes/ros2_control_test_nodes/publisher_joint_trajectory_controller.py

```diff
@@ -28,16 +29,39 @@ def __init__(self):
         self.declare_parameter('wait_sec_between_publish', 6)
         self.declare_parameter('goal_names', ['pos1', 'pos2'])
         self.declare_parameter('joints')
+        self.declare_parameter('check_starting_point', False)
+        self.declare_parameter('starting_point_limits')
 
         # Read parameters
         controller_name = self.get_parameter('controller_name').value
         wait_sec_between_publish = self.get_parameter('wait_sec_between_publish').value
         goal_names = self.get_parameter('goal_names').value
         self.joints = self.get_parameter('joints').value
+        self.check_starting_point = self.get_parameter('check_starting_point').value
+        self.starting_point = {}
 
         if self.joints is None or len(self.joints) == 0:
             raise Exception('"joints" parameter is not set!')
 
+        # starting point stuff
+        if self.check_starting_point:
+            # declare nested params
+            for name in self.joints:
+                param_name_tmp = 'starting_point_limits' + '.' + name
+                self.declare_parameter(param_name_tmp)
+                self.starting_point[name] = self.get_parameter(param_name_tmp).value
+
+            for name in self.joints:
+                if len(self.starting_point[name]) != 2:
+                    raise Exception('"starting_point" parameter is not set correctly!')
+            self.joint_state_sub = self.create_subscription(
+                JointState,
+                'joint_states',
+                self.joint_state_callback,
+                10)
+        self.starting_point_ok = not self.check_starting_point
```

Comment:
Or why not add:

```suggestion
        if not self.check_starting_point:
            self.starting_point_ok = true
        else:
            self.starting_point_ok = false
```

This is much cleanar logic.

Reply:
It can't be set to `None`, because the logic in the repetitive loop is based on it, and it is directly (inverted) connected with the `check_starting_point `variable. Will set it in the proposed - cleaner way.